packages/ai-api/src/utils/deployment-resolver.ts

```diff
@@ -136,7 +130,40 @@ export async function resolveDeploymentId(
       `No deployment matched the given criteria: ${JSON.stringify(opts)}. Make sure the deployment is successful, as it is a prerequisite before consuming orchestration or foundation models.`
     );
   }
-  return deployments[0].id;
+  return deployments[0];
+}
+
+/**
+ * Query the AI Core service for a deployment that matches the given criteria.
+ * If more than one deployment matches the criteria, the first one's ID is returned.
+ * @param opts - The options for the deployment resolution.
+ * @returns A promise of a deployment, if a deployment was found, fails otherwise.
+ * @internal
+ */
+export async function resolveDeploymentId(
+  opts: DeploymentResolutionOptions
+): Promise<string> {
+  const cachedDeployment = deploymentCache.get(opts);
+  if (cachedDeployment?.id) {
+    return cachedDeployment.id;
+  }
+  return (await resolveDeployment(opts)).id;
+}
+
+/**
+ * Query the AI Core service for a deployment that matches the given criteria.
+ * If more than one deployment matches the criteria, the first one's URL is returned.
+ * @param opts - The options for the deployment resolution.
+ * @returns A promise of the deployment URL, if a deployment was found, fails otherwise.
+ */
+export async function resolveDeploymentUrl(
+  opts: DeploymentResolutionOptions
+): Promise<string | undefined> {
```

Comment:
[q] Can the url be undefined ? I haven't checked the actual type that's returned.